### PR TITLE
fix(fault_injection): fix launch_testing

### PR DESCRIPTION
**Pull request by KeisukeShima**
Signed-off-by: Keisuke Shima <19993104+KeisukeShima@users.noreply.github.com>

## Description(required)

Fixed an error in `launch_testing`.
The problem was in the test code, which was testing the state after initialization and the state after receiving a message separately.
Therefore, I decided to work around the issue by combining the test suites into one.

## Review Procedure(required)

Ensure that the CI passes.

## Related PR(optional)

https://github.com/tier4/autoware.iv/pull/2088

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by KeisukeShima** [comment]
https://github.com/autowarefoundation/autoware.universe/runs/5445008216?check_suite_focus=true
It appears that launch_testing is not running.
Hmmm, I guess I need to add some sort of label.

Related:
https://github.com/autowarefoundation/autoware-github-actions/blob/main/colcon-test/action.yaml#L94

**Comment by KeisukeShima** [comment]
Test label for `launch_testing` will be added in ros2/launch 0.21.1. https://github.com/ros2/launch/commit/f40029f2b42d72a17113d318f444261331c9c281#diff-12d0b0fc99d046d60698b3b304d5579778fe7a6006cf39ef27ed6adee9853870R7
In the meantime, I will add custom labels.

File: simulator/fault_injection/CMakeLists.txt [hunk 1]
```diff
@@ -42,16 +42,23 @@ if(BUILD_TESTING)
...
-  #   test/config
-  #   test/launch
-  #   DESTINATION share/${PROJECT_NAME}/
-  # )
+  find_package(launch_testing_ament_cmake REQUIRED)
+  add_launch_test(
+    test/test_fault_injection_node.test.py
+    TARGET test_fault_injection_launch_test
+  )
+  set_tests_properties(
```
Comment (kenji-miyake) [inline simulator/fault_injection/CMakeLists.txt:51]: How about adding a comment here such as `TODO: Remove in Humble`?

Comment (KeisukeShima) [reply simulator/fault_injection/CMakeLists.txt:52]: Good. Added a comment : 80480e333

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**